src/triton_object_recognition/src/object_recognizer.cpp

```diff
@@ -0,0 +1,313 @@
+        *net_ = readNet(model_weights.string(),model_config.string());
+        net_->setPreferableBackend(backend_);
+        net_->setPreferableTarget(target_);
+    }
+
+    void ObjectRecognizer::subscriberCallback(const sensor_msgs::msg::Image::ConstSharedPtr & msg) const
+    {
+        publisher_->publish(process(*msg));
+    }
+
+    void ObjectRecognizer::serviceCallback(const triton_interfaces::srv::ObjectDetection::Request::SharedPtr request, 
+            const triton_interfaces::srv::ObjectDetection::Response::SharedPtr response) const
+    {
+        response->boxes_out = process(request->image_in);
+    }
+
+    triton_interfaces::msg::DetectionBoxArray ObjectRecognizer::process(const sensor_msgs::msg::Image & msg) const
+    {
+        //Get image from message
+        RCLCPP_INFO(this->get_logger(), "In object_recognizer");
+        cv_bridge::CvImagePtr cv_ptr;
+        try {
+            cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
+        } catch (cv_bridge::Exception& e) {
+            RCLCPP_ERROR(get_logger(),"cv_bridge exception: %s", e.what());
+            return triton_interfaces::msg::DetectionBoxArray();
+        }
+
+        //Detect objects in image
+        preprocess(cv_ptr->image);
+        std::vector<cv::String> outNames = net_->getLayerNames();;
+        std::vector<cv::Mat> outs;
+        net_->forward(outs,outNames);
+
+        std::vector<int> classIds;
+        std::vector<float> confidences;
+        std::vector<Rect> boxes;
+        postprocess(classIds,confidences,boxes,cv_ptr->image, outs);
+
+        // Visualize when debugging
+        if (DEBUG_VISUALIZE)
+        {
+            auto drawPred = [&](int classId, float conf, int left, int top, int right, int bottom, Mat& frame)
+            {
+                rectangle(frame, Point(left, top), Point(right, bottom), Scalar(0, 255, 0));
+
+                std::string label = format("%.2f", conf);
+                if (!classes_.empty())
+                {
+                    CV_Assert(classId < (int)classes_.size());
+                    label = classes_[classId] + ": " + label;
+                }
+
+                int baseLine;
+                Size labelSize = getTextSize(label, FONT_HERSHEY_SIMPLEX, 0.5, 1, &baseLine);
+
+                top = max(top, labelSize.height);
+                rectangle(frame, Point(left, top - labelSize.height),
+                        Point(left + labelSize.width, top + baseLine), Scalar::all(255), FILLED);
+                putText(frame, label, Point(left, top), FONT_HERSHEY_SIMPLEX, 0.5, Scalar());
+            };
+            cv::Mat frame = cv_ptr->image;
+            for (size_t idx = 0; idx < boxes.size(); ++idx)
+            {
+                Rect box = boxes[idx];
+                drawPred(classIds[idx], confidences[idx], box.x, box.y,
+                         box.x + box.width, box.y + box.height, frame);
+            }
+            cv::imshow("Object",frame);
```

Comment:
Maybe instead of this, you can have an image_transport publisher which publishes these images. We can still easily see them by using rqt image_view